### PR TITLE
Throttle spatial rotation updates

### PR DESCRIPTION
**Pull request by davedolben**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Add throttling to spatial rotation component updates. Only update if the delta is larger than 0.1 radian (~5.7 degrees).

#### Tests
Tested manually with a large local deployment in PIE with 2500 entities on Bridge v2 with 2 characters. Manually verified that rotation component is still updated, and that significantly fewer rotation updates are sent (drops to almost 0 because most actors in the test map are static).

#### Documentation
n/a

#### Primary reviewers
@girayimprobable @Vatyx

### CODE REVIEW

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -621,6 +621,17 @@ void USpatialActorChannel::UpdateSpatialPosition()
 
 void USpatialActorChannel::UpdateSpatialRotation()
 {
+	FRotator ActorSpatialRotation = Actor->GetActorRotation();
+
+	// Check that it's rotated sufficiently far to be updated.
```
Comment (Vatyx) [inline SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: `Check that the Actor has rotated enough to be updated`

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 2]
```diff
@@ -621,6 +621,17 @@ void USpatialActorChannel::UpdateSpatialPosition()
...
+	const float SpatialRotationThreshold = 0.1f;  // 0.1 radian (~5.7 degrees)
+	FRotator RotationDelta = ActorSpatialRotation - LastSpatialRotation;
+	if (RotationDelta.Quaternion().AngularDistance(FQuat::Identity) < SpatialRotationThreshold)
```
Comment (girayimprobable) [inline SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Couldn't you use `FRotator::Equals()` with a small tolerance parameter to make this simpler?

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 3]
```diff
@@ -621,6 +621,17 @@ void USpatialActorChannel::UpdateSpatialPosition()
 
 void USpatialActorChannel::UpdateSpatialRotation()
 {
+	FRotator ActorSpatialRotation = Actor->GetActorRotation();
+
+	// Check that it's rotated sufficiently far to be updated.
+	const float SpatialRotationThreshold = 0.1f;  // 0.1 radian (~5.7 degrees)
+	FRotator RotationDelta = ActorSpatialRotation - LastSpatialRotation;
```
Comment (danielimprobable) [inline SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: const FRotator

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp (hunk 2 above)
Comment (davedolben) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Looks like this doesn't quite do the same thing. `FRotator::Equals()` checks to see if the difference in any individual component of the rotator (Yaw, Pitch, Roll) is larger than the tolerance parameter, which is much more fuzzy. Happy to change to that if you feel strongly about it, but it doesn't do the same thing.

That said, the current implementation is more expensive, so if it turns out to be a big CPU cycle bottleneck then happy to change to `FRotator::Equals()` later.

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 4]
```diff
@@ -621,6 +621,17 @@ void USpatialActorChannel::UpdateSpatialPosition()
...
+	if (RotationDelta.Quaternion().AngularDistance(FQuat::Identity) < SpatialRotationThreshold)
+	{
+		return;
+	}
+
+	LastSpatialRotation = ActorSpatialRotation;
```
[the remark this answers is not in the document]
Comment (davedolben) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Done, also with LastSpatialPosition for completeness.

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp (hunk 3 above)
Comment (davedolben) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Replaced with a quaternion that gets normalized in place, so can't be const.

**Pull request merged**